File: scenarios/app-configuration-python-config-values/golden/app_configuration.py
import os

from azure.appconfiguration import (
    AzureAppConfigurationClient,
    ConfigurationSetting,
    FeatureFlagConfigurationSetting,
)
from azure.core.exceptions import HttpResponseError
from azure.identity import DefaultAzureCredential


def main() -> None:
    endpoint = os.environ["AZURE_APPCONFIGURATION_ENDPOINT"]
    credential = DefaultAzureCredential()

    try:
        with AzureAppConfigurationClient(
            base_url=endpoint,
            credential=credential,
        ) as client:
            key = "app:Settings:FontSize"
            client.set_configuration_setting(
                ConfigurationSetting(key=key, value="24")
            )
            client.set_configuration_setting(
                ConfigurationSetting(
                    key=key,
                    value="24",
                    label="Production",
                )
            )

            setting = client.get_configuration_setting(key=key)
            print(setting.value)

            for matching_setting in client.list_configuration_settings(
                key_filter="app:Settings:*"
            ):
                print(f"{matching_setting.key}={matching_setting.value}")

            client.set_configuration_setting(
                FeatureFlagConfigurationSetting(
                    feature_id="BetaFeature",
                    enabled=True,
                )
            )
            client.delete_configuration_setting(key=key)
    except HttpResponseError as exception:
        print(f"App Configuration request failed: {exception}", file=os.sys.stderr)
        raise
    finally:
        credential.close()


if __name__ == "__main__":
    main()

File: scenarios/app-configuration-java-config-values/golden/src/main/java/com/example/AppConfiguration.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.data.appconfiguration.ConfigurationClient;
import com.azure.data.appconfiguration.ConfigurationClientBuilder;
import com.azure.data.appconfiguration.models.ConfigurationSetting;
import com.azure.data.appconfiguration.models.FeatureFlagConfigurationSetting;
import com.azure.data.appconfiguration.models.SettingSelector;

public final class AppConfiguration {
    private AppConfiguration() {
    }

    public static void main(String[] args) {
        String connectionString =
                requireEnvironment("AZURE_APPCONFIG_CONNECTION_STRING");
        ConfigurationClient client = new ConfigurationClientBuilder()
                .connectionString(connectionString)
                .buildClient();
        String key = "app:Settings:FontSize";

        try {
            client.setConfigurationSetting(key, null, "24");
            client.setConfigurationSetting(key, "Production", "24");

            ConfigurationSetting setting =
                    client.getConfigurationSetting(key, null);
            System.out.println(setting.getValue());

            SettingSelector selector =
                    new SettingSelector().setKeyFilter("app:Settings:*");
            client.listConfigurationSettings(selector).forEach(item ->
                    System.out.println(item.getKey() + "=" + item.getValue()));

            client.setConfigurationSetting(
                    new FeatureFlagConfigurationSetting("BetaFeature", true));
            client.deleteConfigurationSetting(key, null);
        } catch (HttpResponseException exception) {
            System.err.printf(
                    "App Configuration request failed with status %d: %s%n",
                    exception.getResponse().getStatusCode(),
                    exception.getMessage());
            System.exit(1);
        }
    }

    private static String requireEnvironment(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException("Set " + name + " before running.");
        }
        return value;
    }
}
